microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/core/utils/AnnotationUtils.java

```diff
@@ -68,7 +74,85 @@ public static IAnnotation getAnnotation(IAnnotatable annotatable, String annotat
 	 *         false otherwise.
 	 */
 	public static boolean isMatchAnnotation(IAnnotation annotation, String annotationName) {
-		return annotationName.endsWith(annotation.getElementName());
+		// Annotation name is the fully qualified name of the annotation class (ex :

```

Comment:
Could you please review this method @rgrunber ?

Reply:
I think it looks good. I guess it's worth doing it this way to avoid creating new import objects every time getImports() is called ?

Also, I don't think this would handle on-demand imports (eg. `org.eclipse.microprofile.config.inject.*` ).

Reply:
> I guess it's worth doing it this way to avoid creating new import objects every time getImports() is called ?

exactly

> Also, I don't think this would handle on-demand imports (eg. org.eclipse.microprofile.config.inject.* ), but it should still be a good improvement even without.

You are right. I'm not sure it's a commons usecase. We could support that if we have an user issue. What do you think?

Reply:
That's fine with me. If it isn't common we can ignore for now. Although I think eventually, something like :
```
if (importDeclaration.isOnDemand()) {
  String fqn = importDeclaration.getElementName();
  String qualifier = fqn.substring(0,fqn.lastIndexOf('.'));
  return qualifier.equals(annotationName.substring(0,annotationName.lastIndexOf('.')));
}
```
should work. The initial check would also need to exclude returning false when the ending name of the import doesn't match the qualified name if ".*" is present.